src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -302,8 +309,20 @@ class MariaDBManager implements IQueryService {
         int personID = rs.get("responsible_person_id");
         Person pers = getPersonNameByID(personID)
 
-        if(currID == locID) {
-          currLoc = new Location(name: name, responsiblePerson: pers.getFirstName()+" "+pers.getLastName(), responsibleEmail: pers.getEMail(), address: address, status: status, arrivalDate: arrivalDate, forwardDate: forwardedDate);
+        if(currID == locID ) {
+           // Compare if current location is the newest entry
+            if (currDate && currDate.before(arrivalDate)) {
```

Comment:
```suggestion
            if (currDate && currDate?.before(arrivalDate)) {
```

Reply:
Groovy truth: the first conditional expression evaluates to false, if `currDate` is `null`. The second expression will not be executed, thus no NullPointerException.

Reply:
Ah very good! Thanks for the explanation!